### PR TITLE
[Spec] Remove unused CUDA draft-extend top-k from graph

### PR DESCRIPTION
**Pull request by mattteochen**
## Motivation

DRAFT_EXTEND_V2 CUDA graphs compute full-row top-k results that are discarded after replay. The worker selects the correct logits row and recomputes the consumed top-k values, so the captured memset, argmax reduction, and fill add unnecessary work.

## Modifications

- Skip the unused graph-local top-k computation on CUDA while preserving the existing ROCm path.
- Align the eager graph-runner reference with the DRAFT_EXTEND_V2 output contract.

## Accuracy Tests

main:
```bash
== gsm8k ==
1319 examples (single-shot)  |  541.8s  |  2056 tok/s  |  1.1M tokens

* score           =  94.24%
  stop_rate       =  96.29%
  truncated_rate  =  3.71%  [warn: hitting max_tokens]
  error_rate      =  0.00%
```

PR:
```bash
== gsm8k ==
1319 examples (single-shot)  |  566.6s  |  1928 tok/s  |  1.1M tokens

* score           =  94.39%
  stop_rate       =  96.36%
  truncated_rate  =  3.64%  [warn: hitting max_tokens]
  error_rate      =  0.00%
```

## Speed Tests and Profiling

main:
image

PR:
image

cc @kpham-sgl 

---
### CI States

Latest PR Test (Base): :x: [Run #29491452451](https://github.com/sgl-project/sglang/actions/runs/29491452451)
Latest PR Test (Extra): :x: [Run #29491452286](https://github.com/sgl-project/sglang/actions/runs/29491452286)

### CODE REVIEW

**Comment by nvpohanh** [comment]
@mattteochen is this dup of #31430 ?

**Comment by mattteochen** [comment]
> @mattteochen is this dup of #31430 ?

@nvpohanh oh yes, didn't see that, thank you.
